run_dir/static/js/project_samples.js

```diff
@@ -352,12 +375,19 @@ function load_all_udfs(){
         }
       }
 
-      // Make the project emails clickable, and add labels.
+      // Make the project emails clickable and add labels.
       else if (prettify(key) == 'contact'){
         function elabel(text, label) {
           return '<span class="label label-'+label+'">'+text+'</span>'
         }
-        if(data['order_details']) {
+        // Stolen, without shame, from https://stackoverflow.com/a/46181
+        function validateEmail(email) {
+          var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+          return re.test(String(email).toLowerCase());
```

Comment:
Should we try get any email first from given string ? and if we couldn't then return false ? something like

```
function validateEmail(email) {
   var cap_email = null;
   var re = /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))/;
   if (email){
        matches = email.match(re);
        if (matches){
           cap_email = matches[0];
        }
   }
   return cap_email;
}
```

Downstream we could use the validated and clean email. Well if you think this is a overkill or might have unintended effect in some cases we could just go with the more stringent way 👍

Reply:
I like this idea. 👍 
I would have to test it on stage first.